test/Model/UserSpec.hs

```diff
@@ -82,6 +82,56 @@ spec = withApp $ do
                 runDB (authenticateUser' creds')
                     `shouldReturn` ServerError "github: missing key email"
 
+        context "from Google" $ do
+            let creds = Creds
+                    { credsPlugin = "googleemail2"
+                    , credsIdent = "1"
+                    , credsExtra =
+                        [ ("name", "{\"formatted\":\"Jim Smith\"}")
+                        , ("emails", "[{\"value\":\"jim@gmail.com\",\"type\":\"account\"}]")
+                        ]
+                    }
+
+            it "creates a user from the name/emails values" $
+                creds `shouldCreateWith` Profile "Jim Smith" "jim@gmail.com"
+
+            it "updates an existing user from the name/emails values" $
+                creds `shouldUpdateWith` Profile "Jim Smith" "jim@gmail.com"
+
+            it "errors if name is missing" $ do
+                let creds' = creds { credsExtra = [("email", "[{\"value\":\"\",\"type\":\"account\"")] }
+
+                runDB (authenticateUser' creds')
+                    `shouldReturn` ServerError "googleemail2: missing key name"
+
+            it "errors if name has no values" $ do
+                let creds' = creds
+                        { credsExtra =
+                            [ ("name", "{}")
+                            , ("email", "[{\"value\":\"\",\"type\":\"account\"")
+                            ]
+                        }
+
+                runDB (authenticateUser' creds')
+                    `shouldReturn` ServerError "googleemail2: user has no name"
```

Comment:
This is handled by the change I made in `yesod-auth`. The behavior depends on what we return:

- `Authenticated user` - all OK
- `UserError msg` - Show `msg` to the user (it's always "Invalid Login" for now)
- `ServerError err` - Log `err` and show the user "Authentication failure"